src/mds/Server.cc

```diff
@@ -4377,13 +4377,14 @@ void Server::handle_client_open(MDRequestRef& mdr)
     respond_to_request(mdr, -CEPHFS_EROFS);
     return;
   }
-  
+
   CInode *cur = rdlock_path_pin_ref(mdr, need_auth);
   if (!cur)
     return;
 
-  if (cur->is_frozen() || cur->state_test(CInode::STATE_EXPORTINGCAPS)) {
-    ceph_assert(!need_auth);
+  // In case the _open() is called from _openc() the previous rdlock_path_pin_ref()
+  // will fail to forward the request to auth MDS, we need to retry it here.
```

Comment:
Can you expand in the comment here why this particular flow will fail to forward the request to auth?

I feel we could write a test for this:

- make a config for the client to always choose a particular rank by default
- make the necessary file, pin the directory its in to rank 1
- set the config to have the client always choose rank 0 by default
- write a python script to open the file `O_CREATE|O_TRUNC|O_WRONLY`.

yes?

Reply:
Certainly, I will expand the comment and try to add a test case for it. Thanks @batrick